File: backend/app/schemas/sosovalue.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class SoSoValueEventResponse(BaseModel):
    id: UUID
    event_type: str
    source_id: str
    published_at: datetime
    symbols: list[str] | None
    sentiment_score: float | None = None
    importance_score: float | None = None
    title: str | None = None
    summary: str | None = None
    payload: dict | None = None

    model_config = {"from_attributes": True}


class MarketContextResponse(BaseModel):
    symbol: str
    pair_market: dict | None
    recent_news: list[SoSoValueEventResponse]
    macro_events: list[SoSoValueEventResponse]
    current_regime: str
    regime_confidence: float

File: frontend/src/lib/api/sosovalue.ts
import { api } from '@/lib/api/client'

export interface SoSoValueEvent {
  id: string
  event_type: string
  source_id: string
  published_at: string
  symbols: string[] | null
  sentiment_score: number | null
  importance_score: number | null
  title: string | null
  summary: string | null
  payload: Record<string, unknown> | null
}

export interface MarketContext {
  symbol: string
  pair_market: Record<string, unknown> | null
  recent_news: SoSoValueEvent[]
  macro_events: SoSoValueEvent[]
  current_regime: string
  regime_confidence: number
}

export async function getSoSoValueEvents(eventType?: string): Promise<SoSoValueEvent[]> {
  const query = eventType ? `?event_type=${eventType}` : ''
  return api.get(`/api/v1/sosovalue/events${query}`)
}

export async function getMarketContext(symbol = 'BTC'): Promise<MarketContext> {
  return api.get(`/api/v1/sosovalue/context?symbol=${symbol}`)
}
